app/controllers/solidus_paypal_braintree/client_tokens_controller.rb

```diff
@@ -18,5 +16,20 @@ def load_gateway
         @gateway = ::SolidusPaypalBraintree::Gateway.where(active: true).merge(store_payment_methods_scope).first!
       end
     end
+
+    def generate_token
+      options = {}
+      options[:customer_id] = customer_id if customer_id.present?
+
+      @gateway.generate_token(options)
+    rescue ::SolidusPaypalBraintree::Gateway::TokenGenerationDisabledError => error
+      Rails.logger.error error
+      nil
+    end
+
+    def customer_id
+      return unless try_spree_current_user&.braintree_customer
```

Comment:
There is a bug with the `ClientTokensController`. The controller generates the token for the current user. However if that endpoint is called from admin then it will attach the payment source to the administrating user rather than the user on the order

Reply:
good catch!

Reply:
Debating if the token should be generated when the view is rendered. Maybe as a hidden input

Reply:
Not sure, what's the difference?

Reply:
See implementation 2 here: https://github.com/solidusio/solidus_braintree/pull/79

The main difference is that the view no longer makes an api request to obtain the client token but rather just generates it when the view renders.  The api request was only made if the new braintree payment method was selected (javascript events).  This change generates a token even if the user may not be adding a new payment method. Therefore it could mean a slight increase in api requests sent to braintree. I think this should be negligible though

Reply:
What about who is using this gem without a Rails frontend, eg on React or Vue.js? I think we still need this controller to work for them, right?